src/pages/ProfilesExplorerView/components/SceneByVariableRepeaterGrid/SceneByVariableRepeaterGrid.tsx

```diff
@@ -392,96 +435,137 @@ export class SceneByVariableRepeaterGrid extends SceneObjectBase<SceneByVariable
       return;
     }
 
-    const { headerActions } = this.state;
-
     const gridItems = this.state.items.map((item) => {
-      const gridItemKey = SceneByVariableRepeaterGrid.buildGridItemKey(item);
-      const { queryRunnerParams } = item;
-
-      const data = new SceneDataTransformer({
-        $data: buildTimeSeriesQueryRunner(queryRunnerParams),
-        transformations: [SceneByVariableRepeaterGrid.limitNumberOfSeries],
-      });
-
-      const timeSeriesPanel = PanelBuilders.timeseries()
-        .setTitle(item.label)
-        .setData(data)
-        .setHeaderActions(headerActions(item))
-        .build();
+      const vizPanel: SceneObject = this.buildVizPanel(item);
 
       if (this.state.hideNoData) {
-        this.setupHideNoData(timeSeriesPanel);
+        this.setupHideNoData(vizPanel);
       }
 
-      this.setupOverrides(timeSeriesPanel, item);
-
       return new SceneCSSGridItem({
-        key: gridItemKey,
-        body: timeSeriesPanel,
+        key: SceneByVariableRepeaterGrid.buildGridItemKey(item),
+        body: vizPanel,
       });
     });
 
     (this.state.body as SceneCSSGridLayout).setState({
-      autoRows: GRID_AUTO_ROWS, // required to have the correct grid items height
+      autoRows: this.getAutoRows(), // required to have the correct grid items height
       children: gridItems,
     });
   }
 
-  setupOverrides(timeSeriesPanel: VizPanel, item: GridItemData) {
-    const sub = (timeSeriesPanel.state.$data as SceneDataTransformer)!.subscribeToState((state) => {
+  getAutoRows() {
+    const { variableName } = this.state;
+
+    if (variableName === 'groupBy' && getSceneVariableValue(this, variableName) !== 'all') {
+      const panelTypeSwitcher = findSceneObjectByClass(this, ScenePanelTypeSwitcher) as ScenePanelTypeSwitcher;
+
+      return panelTypeSwitcher.state.type === PanelType.BARGAUGE ? GRID_AUTO_ROWS_SMALL : GRID_AUTO_ROWS;
```

Comment:
We reduce the height of the panels because we'll display a single bar:
<img width="1701" alt="image" src="https://github.com/user-attachments/assets/78e66065-77bc-4208-b77a-9ece73d567f1">

But we might have to reconsider in the near future if we want display bar gauges after selecting a group by value.

Reply:
I think bars only make sense when on the same scale (i.e. I wouldn't separate these individually here and instead just put them on one graph